### ISSUE TITLE
Add Previous events to event page

### ISSUE THREAD

**Issue by davehakkens** [ONEARMY/community-platform#1705]
### Description

We would like to display previous events in the event module.
- It makes the page more rich
- It could help people to find eachother locally even after events.

### Page related

Will be used in : [Event page](https://community.preciousplastic.com/events)

### Build suggestion
- Previous events should be greyed out. 
- They are loaded with infinite scroll as the [how-to](https://community.preciousplastic.com/how-to) page)

### Mockup
![events-previous](https://user-images.githubusercontent.com/13672737/171494947-9ee8a098-7977-4e11-aef9-80866a82b858.jpg)

**Comment by Xyli0** [comment]
Hi, everyone,
I just found your project and would love to contribute.
I am all set and would like to work on this issue if you don't mind 😃.
@mattia-io

**Comment by thisislawatts** [comment]
Welcome aboard @Xyli0, let us know if you have any questions getting up and running with this issue.

**Comment by Xyli0** [comment]
Thanks @thisislawatts. I start working on it.

**Comment by Xyli0** [comment]
I took a look at the code and I am still trying to figure out how to implement infinite scrolling.

For now I suggest to put the two lists into their own component named `UpcomingEventsList` and `PreviousEventsList` located inside /src/pages/Events/content/ and rename _/src/pages/Events/Content/`EventList`_ to `EventLists` to display both components.

Currently `Eventlist` displays the list of upcoming events if `filteredEvents` is true else it displays `<div>Events not found</div>`. I suggest that for the upcoming events list if there is no data found to display the "create an event" CTA. 

![Screenshot 2022-06-04 at 12-00-31 Precious Plastic Community](https://user-images.githubusercontent.com/10441748/171994573-16b0b0aa-8949-40e8-91ec-523a7dc2ede3.png)

So for instance if there is no filtered upcoming events it would look like this (don't pay attention to the scale)
![community_platform_list_mockup](https://user-images.githubusercontent.com/10441748/171996094-35fb0e42-9042-47fa-a604-90105ac11a1e.png)

For the infinite scrolling i tried some things with react-virtualized  but I am not sure if it is the way to follow. I used `WindowScroller` but I get weird behaviours with the layout while scrolling. Maybe I could use `InfiniteLoader` and load (let's say) five events every time  the scroll hits the bottom of the page. I am not sure...

What do you think?

**Comment by thisislawatts** [comment]
Hey @Xyli0,

Thanks for raising this point. I actually think that we should not use infinite scrolling here and instead introduce pagination. I am aware that we currently use infinite loading on the how-to listing but this is something we will likely remove to improve the SEO performance. 

So maybe the block in the design shows the 20 most recent past events and then we have a link through to an archive page.

For example:
https://community.preciousplastic.com/events/past

I would need to check how many events we have in the system at the moment, but if it's less than 200 I think we can do without pagination in the immediate and add that in a future iteration.

**Comment by thisislawatts** [comment]
@ONEARMY/design, @Xyli0 has flagged an additional state here. When there are no upcoming events and _only_ past events. What is the most useful thing we can present our users with in that situation?

**Comment by davehakkens** [comment]
yeh thanks for adding that state @Xyli0, sounds good!
Altough I would propose another text for that state, something like:
"At the moment there are no upcoming events" 
(don't display "No data to show")

<img width="524" alt="afbeelding" src="https://user-images.githubusercontent.com/13672737/172018737-e29119e2-e9e3-4807-8c88-44cb68a5068a.png">

Also up for swapping that infinite scrolling for pagination for better SEO in the future🔥
And agree with @thisislawatts we can just skip pagination for now and display all since it isn't too much content..

**Comment by Xyli0** [comment]
Thanks @thisislawatts @davehakkens, I know enough to continue.

**Comment by Xyli0** [comment]
Hi everyone,

I have been able to display a list of past events by reusing `src/components/EventCard/EventCard.tsx` (not greyed out). 

![Screenshot 2022-06-08 at 11-18-42 Community Platform](https://user-images.githubusercontent.com/10441748/172586179-3f3db528-3be9-4887-a8cf-cc8c3bc9b850.png)

So now I have to grey these items out.
To proceed I need to clarify two things:

Should the "Awaiting moderation" badge (see picture) stay in place for past events. I guess no because old events should have already been moderated before they take place.

I would like to add a prop like called `inactive` or `disabled` on `EventCard` to make the component grey out when true. It avoids creating a new component only for greyed out items. 
What do you think of this option ? Maybe positive values would be cleaner ( like `active)`

Thanks

ps: I was slowed up by holidays here, now I'm back home, I'll be able to deliver soon...

**Comment by davehakkens** [comment]
Good to see this moving forward!

I think you could keep the "awaiting moderation" badges visable. 
This is only shown to admins and indeed should not be there in the first place.
But if there is one that slipped through its a good moment to revisit and moderate it. (for the users point system)

The second question i'll leave up to devs:)

**Comment by Xyli0** [comment]
The feature is pretty much done however I am a bit stuck for now with testing.

I wrote a test in 'events.spec.ts` but it seems that there is no previous events stored in the test database so my test can't pass in the current state.
Could somebody add a bunch of past events to the database or should I skip it for now? 

Also while looking at the tests I found a step that doesn't look right. In `events.spec.ts` line 15 & 16:

``` 
cy.step('Move Events button is hidden')
cy.get('button').contains('More Events').should('not.visible')
 ```
it looks for a hidden button that always remains hidden no matter what located in `/EventsList.tsx` line 124 to 130:
```
<Flex sx={{ justifyContent: 'center' }} mt={20}>
  <Link to={'#'} style={{ visibility: 'hidden' }}>
    <Button variant={'secondary'} data-cy="more-events">
      More Events
    </Button>
  </Link>
</Flex>
```

Maybe I am missing something but I think this button has no purpose and so those two lines of test. Could somebody confirm ?

I reused the button in my code to be sure not to break anything for now but if it has no usefulness I will remove it before PR.

Thanks
